### PR TITLE
Remove soft break tofu in cairo

### PR DESCRIPTION
**Pull request by derekdreery**
I still need to figure out how to regenerate the snapshots.

### CODE REVIEW

**Comment by derekdreery** [comment]
It might be that this has to wait until proper shaping, but I will try to land something sensible sooner.

File: piet-cairo/src/text/lines.rs
Comment (cmyr) [inline piet-cairo/src/text/lines.rs:137]: Was slightly confused here for a second; I couldn't tell that this wouldn't screw up in cases where there were multiple newlines in a row, which should be treated as actual newlines.

Since this is linux I'm also not sure we care about `\r`?

just for fun I'd have probably written something like,

```rust
let line_text = &text[line_start..line_end];
let newline_len = line_text.as_bytes().rev().take_while(|b| b == b'\n'').count();
let line_end = line_end - newline_len;
```

**Review by cmyr — APPROVED** [review]
Cool, definitely worth getting this cleaned up.

There are instructions for adding snapshots somewhere, I think in `CONTRIBUTING.md`. Ping me on zulip if you need help!

File: piet-cairo/src/text/lines.rs
Comment (derekdreery) [reply piet-cairo/src/text/lines.rs:137]: Yeah `\r` is probably overkill.

Comment (derekdreery) [reply piet-cairo/src/text/lines.rs:137]: I also think your solution is better.

**Comment by cmyr** [comment]
going to close this since we'll resolve in #389.

**Pull request closed without merging**
